python/grass/jupyter/display.py

```diff
@@ -0,0 +1,101 @@
+
+# MODULE:    grass.jupyter.display
+#
+# AUTHOR(S): Caitlin Haedrich <caitlin DOT haedrich AT gmail>
+#
+# PURPOSE:   This module contains functions for non-interactive display
+#            in Jupyter Notebooks
+#
+# COPYRIGHT: (C) 2021 Caitlin Haedrich, and by the GRASS Development Team
+#
+#           This program is free software under the GNU General Public
+#           License (>=v2). Read the file COPYING that comes with GRASS
+#           for details.
+
+import os
+import folium
+import grass.script as gs
+
+class grassRenderer:
+    """
+    The grassRenderer class creates and displays GRASS maps in Jupyter
+    Notebooks. 
+    """
+
+    def __init__(self, width=600, height=400, text_size=12):
+        self.height = height
+        self.width = width
+        os.environ["GRASS_RENDER_WIDTH"] = str(width)
+        os.environ["GRASS_RENDER_HEIGHT"] = str(height)
+        os.environ["GRASS_TEXT_SIZE"] = str(text_size)
+        gs.run_command("d.erase")
+
+
+    def d_rast(self, raster, color="greys"):
+        """
+        Adds a raster to the display
+        """
+        gs.run_command("r.colors", map=raster, color=color)
```

Comment:
Since this actually modified the raster, this may not be what is expected from the display function. Unfortunately, it also means that you need to do it ahead of time and can't set it here. I'm interested what everybody thinks about this.

Reply:
I tested the notebook, looks nice ;)
Indeed, it seems odd to me to be forced to set a color when d.* modules should just read the defined color table or use the default, setting a color should be optional IMHO

Reply:
Agreed with Vero here...

Reply:
Perhaps the issue is not as much "being forced to set a color" (defaults can be changed) as that the expectation is that you set the color at this point. And perhaps that the reason @chaedri wrote it initially this way. Like for vectors. Not Jupyter specific, but it is more visible here.

I have never tried that, but even with the current system, you could implement ad-hoc/temporary color tables for rasters, I think. You create a temporary mapset, set color table for raster of interest there (raster color tables are per-mapset), display  the raster from that mapset, so the mapset-specific color table kicks in. I'm not suggesting @chaedri should do this :-)